src/sentry/incidents/models/alert_rule.py

```diff
@@ -496,6 +496,59 @@ def get_queryset(self) -> BaseQuerySet[AlertRuleTriggerAction]:
         return super().get_queryset().exclude(status=ObjectStatus.PENDING_DELETION)
 
 
+class AlertRuleActionHandlerFactory(abc.ABC):
```

Comment:
I know this name is something out of an OOP nightmare. I'm open to suggestions.

Reply:
Maybe it can just be `ActionHandlerFactory` since it's already in the `incidents` project and it's specific to that?